ui/app/components/app/modals/account-details-modal.js

```diff
@@ -86,8 +92,10 @@ AccountDetailsModal.prototype.render = function () {
       h(Button, {
         type: 'secondary',
         className: 'account-modal__button',
-        onClick: () => global.platform.openWindow({ url: genAccountLink(address, network) }),
-      }, this.context.t('etherscanView')),
+        onClick: () => {
+          global.platform.openWindow({ url: genAccountLink(address, network, rpcPrefs) })
+        },
+      }, this.context.t('blockExplorerView', rpcPrefs.blockExplorerUrl && [rpcPrefs.blockExplorerUrl])),
```

Comment:
I am getting `View account at $1` on the modal screen when on the default Ethereum networks.  Could we do 
```suggestion
      }, rpcrpcPrefs.blockExplorerUrl ? this.context.t('blockExplorerView [rpcPrefs.blockExplorerUrl])) : this.context.t('viewOnEtherscan'),
```
and with `[rpcPrefs.blockExplorerUrl]` could we parse the url with Regex for just the domain? For my first ugly stab at it, 
`customBlockExplorer.match(regex)[0].charAt(0).toUpperCase() + customBlockExplorer.match(regex)[0].slice(1).replace(/[^a-zA-Z ]/gi, '')`

Also, needs the same logic for https://github.com/MetaMask/metamask-extension/blob/31175625b446cb5d18b17db23018bca8b14d280c/ui/app/components/app/dropdowns/account-details-dropdown.js#L118

Reply:
Fixed in ed81fed19